### PR TITLE
GeoDataFrame.from_url()

### PR DESCRIPTION
**Pull request by dzbee**
Added method to GeoDataFrame for ingesting GeoJSON files that are hosted online (for example https://data.cityofnewyork.us/resource/kk4q-3rt2.geojson). This is intended to provide similar functionality to `pandas.read_json()`. Included a somewhat trivial test against the url for `geopandas/examples/null_geom.geojson` on Github.

There's a bunch of unrelated deletions/insertions where my editor changed things w/ automatic PEP8 adjustments – sorry.

### CODE REVIEW

File: geopandas/tests/test_geodataframe.py [hunk 1]
```diff
@@ -286,6 +288,15 @@ def test_copy(self):
         self.assertTrue(type(df2) is GeoDataFrame)
         self.assertEqual(self.df.crs, df2.crs)
 
+    def test_from_url(self):
+        """ Test from_url """
+        # Write layer with null geometry out to file
```
Comment (jdmcbr) [inline geopandas/tests/test_geodataframe.py]: This comment seems misplaced.

**Comment by jdmcbr** [comment]
This looks like a great addition to me.

File: geopandas/tests/test_geodataframe.py (hunk 1 above)
Comment (dylanbGIT) [inline geopandas/tests/test_geodataframe.py]: Fixed!

**Comment by kuanb** [comment]
Re: those unintended adjustments, it looks like it's because you `--max-line-length` argument may be set to 80 instead of 120 (assuming Geopandas is set to 120 due to the line changes that occur in your PR).

**Comment by jorisvandenbossche** [comment]
@dzbee this indeed looks like a nice addition!

Just an idea about the API: I think it would be nice to integrate this in the existing read functions, instead of creating a specific function for urls (so you can pass an url to the existing functions). 
Because also in principle the url can can hold any type of file, not necessarily a json file.

**Comment by jorisvandenbossche** [comment]
@dzbee Thanks for working on it anyway! (and certainly let know if there are limitations with the merged approach)

**Pull request closed without merging**